packages/components/src/presentation-components/properties/inputs/UniquePropertyValuesSelector.tsx

```diff
@@ -251,27 +247,40 @@ function useUniquePropertyValuesLoader({ imodel, ruleset, field, descriptorInput
             from(Presentation.presentation.getPagedDistinctValues(requestProps)).pipe(map((result) => result.items))
         ).subscribe({
           next: resolve,
-          error: reject,
+          error: () => resolve([]),
         });
       });
 
+      const options: UniqueValue[] = [];
       const filteredOptions: UniqueValue[] = [];
       for (const option of items) {
         if (option.displayValue === undefined || !DisplayValue.isPrimitive(option.displayValue)) {
           continue;
         }
         const groupedValues = option.groupedRawValues.filter((value) => value !== undefined);
-        if (groupedValues.length !== 0) {
-          filteredOptions.push({ displayValue: option.displayValue, groupedRawValues: groupedValues });
+
+        if (groupedValues.length === 0) {
+          continue;
+        }
+
+        const uniqueValue = { displayValue: option.displayValue, groupedRawValues: groupedValues };
+        options.push(uniqueValue);
+
+        if (matchesSearchInput(uniqueValue)) {
+          filteredOptions.push(uniqueValue);
         }
       }
 
+      setLoadedCount((prev) => prev + items.length);
+      setLoadedOptions([...loadedOptions, ...options]);
+      setHasMore(options.length === UNIQUE_PROPERTY_VALUES_BATCH_SIZE);
```

Comment:
I think this should be merged into single `useState` as they are always updated together